server.js

```diff
@@ -20,4 +30,94 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+
+
+//all messages
+app.get("/messages", (request, response) => {
+  response.send(messages)
+})
+
+const dateObject = new Date();
+const hours = dateObject.getHours();
+const minutes = dateObject.getMinutes();
+const seconds = dateObject.getSeconds();
+
+
+
+//send a message
+app.post("/messages", (request, response) => {
+  let newMessage = request.body
+  if (newMessage.from === "" || newMessage.text === "") {
+    console.log(console.log(console.error()))
+    throw new Error("400")
+  } else {
+    newMessage.timeSent = `${hours}:${minutes}:${seconds}`
```

Comment:
I think if you have the hours, minutes and seconds defined outside of this function (and the new date as well), then after the server started to run, it will add the exact same time to all posted messages. To capture the actual creation time, the new date have to be created within the function, so it always reflects the time when the function is called.

Reply:
@GergelyKI Thank you very much, I've fixed it. I'm not sure why bat it shows time 1 hour less than UK TIME.